src/main/java/com/oltpbenchmark/api/Worker.java

```diff
@@ -74,12 +74,14 @@ public Worker(T benchmark, int id) {
         this.currStatement = null;
         this.transactionTypes = this.configuration.getTransTypes();
 
-        try {
-            this.conn = this.benchmark.makeConnection();
-            this.conn.setAutoCommit(false);
-            this.conn.setTransactionIsolation(this.configuration.getIsolationMode());
-        } catch (SQLException ex) {
-            throw new RuntimeException("Failed to connect to database", ex);
+        if (!this.configuration.getNewConnectionPerTxn()) {
+            try {
+                this.conn = this.benchmark.makeConnection();
```

Comment:
Can you refactor `BenchmarkModule.makeConnection()` so that it takes in the autoCommit + txnIsolation level as arguments? That way we don't have dupe code for your new feature.

Reply:
Ignore this statement. I see why you didn't do this.